src/mo_frontend/parser.mly

```diff
@@ -334,7 +337,7 @@ and objblock eo s id ty dec_fields =
 
 seplist(X, SEP) :
   | (* empty *) { [] }
-  | x=X { [x] }
+  | x=X { [x] } [@recover.cost inf]
```

Comment:
Why is this needed?

Reply:
There is a case:
```
let x = 2 // missing ;
let y = 2;
```
Error recovery with default costs is always used to reduce the declaration list and just skip all the rest.
This infinite cost forces it rather go on `x=X SEP xs=seplist(X, SEP) { x::xs }` path and generate SEP (`;`) and then parse the rest. 

It's kind of a hack, but works